tools/bin/gppylib/commands/unix.py

```diff
@@ -925,7 +925,7 @@ def getDescendentProcesses(pid):
     children = list()
     grandchildren = list()
 
-    for p in psi.process.ProcessTable().values():
+    for p in psutil.process.ProcessTable().values():
```

Comment:
Just to close the loop on this. The psi module does not ship for Ubuntu any more (at least out of the box with `pip` or via standard apt repos) from what I can tell with Ubuntu 14.04; it has not been touched since 2010 either. The psutil module is largely API compatible and maintained: shipping for both CentOS and Ubuntu via pip or packages.

For reference, it looks like GPDB has decided to do the same for these  now too: https://github.com/greenplum-db/gpdb/pull/486

Further, I did not find a test suite, so I tried to migrate this as much as possible; looking at the changes the Greenplum community did, clearly I missed a lot of necessary changes here.